src/pages/api/locks/lock_door.ts

```diff
@@ -0,0 +1,33 @@
+import { z } from "zod"
+
+import { withRouteSpec } from "lib/middleware/index.ts"
+import { action_attempt } from "lib/zod/index.ts"
+
+export const jsonBody = z.object({
+  device_id: z.string(),
+  sync: z.boolean().default(false),
+})
+
+export default withRouteSpec({
+  methods: ["POST"],
+  auth: "cst_ak_pk",
+  jsonBody,
+  jsonResponse: z.object({
+    action_attempt,
+  }),
+} as const)(async (req, res) => {
+  const { sync } = req.body
+
+  const action_attempt = req.db.addActionAttempt({
+    action_type: "LOCK_DOOR",
+  })
+
+  const action_attempt_sync = req.db.updateActionAttempt({
+    action_attempt_id: action_attempt.action_attempt_id,
+    status: "success",
+  })
+
+  res.status(200).json({
+    action_attempt: sync ? action_attempt_sync : action_attempt,
+  })
+})
```

Comment:
This should give you the right response back, so it's fine. But I wonder if it's worth it in this repo to introduce the concept of a delayed action like we have in other fakes, that way you could test asynchronous actions properly. 

I think you encountered this a bit in the hubitat fake @mxsdev 

Wouldn't expect all of that in this PR

Reply:
excellent idea, can't wait for this functionality! internal endpoints could help complete action attempts for tight control, but also accelerating time etc. is an option!

Reply:
Yep, this thought occurred to me when I was making this! If I just add a `setTimeout` that updates the DB, will that work in practice? Or do I have to do something more complex?

Anyways, definitely interested in doing this in a follow-up PR.

Reply:
in fakes we should use the `update(t)` function to do time-based updates instead of `setTimeout`! `update(t)` is called prior to any endpoint to ensure that all time-based things happen!